### PR TITLE
Config map view inconsistent for binary data

### PR DESCRIPTION
**Pull request by aalves08**
Addresses Github issue: [#5311](https://github.com/rancher/dashboard/issues/5311)
Addresses Zube issue: [#5336](https://zube.io/rancher/dashboard-ui/c/5336)

- Create dedicated tab to `binaryData` for `configmap`
- Update `configmap` edit component to not encode/decode `binaryData`
- Update  `KeyValue` component to better handle binary data
- Create `binarySize` crypto utils function

**UPDATE:**
- I believed that I've cracked the problems we were having with encoding/decoding of base64, so I restored all the functionalities we previously had... The main change in the `configmap` it the separation between `data` and `binaryData`.

With that in mind, most of the comments in this PR should not be considered when reviewing because they were done when we thought the best of course of action was limiting some actions so that we wouldn't risk compromising `base64` data

### CODE REVIEW

**Review by nwmac — CHANGES_REQUESTED** [review]
Looks good to me.

On the edit config page for a config map, can we add a message to the binary data tab when there is no binary data - at the moment you just get:
image

Something like: "No binary data'.

Also - when there is binary data, can we add an info banner underneath the 'Binary Data' heading that says 'Binary data can not be edited in the UI'

Lastly, although binary data can not be edited, it would nice to be able to remove binary data keys.

**Review by nwmac — CHANGES_REQUESTED** [review]
Two small things:

- Can we change the message to "Binary Data values can not be edited in the UI" = since you can change the key name and delete - you can't change the value
- Can we only show the above message when the mode is edit - currently you see the message about not being able to edit when you are viewing the config

File: components/form/KeyValue.vue [hunk 1]
```diff
@@ -2,49 +2,41 @@
...
-import { asciiLike } from '@/utils/string';
-import { base64Encode, base64Decode } from '@/utils/crypto';
+import { base64Encode, base64Decode, binarySize } from '@/utils/crypto';
 import { downloadFile } from '@/utils/download';
 import TextAreaAutoGrow from '@/components/form/TextAreaAutoGrow';
 import { get } from '@/utils/object';
 import Select from '@/components/form/Select';
 import FileSelector from '@/components/form/FileSelector';
 import { _EDIT, _VIEW } from '@/config/query-params';
-
```
Comment (richard-cox) [inline components/form/KeyValue.vue:13]: I think you might have an odd linting rule that puts these in. Generally I think we have lines where they have been removed in the file

File: assets/translations/en-us.yaml [hunk 2]
```diff
@@ -1714,6 +1714,7 @@ configmap:
       protip: Use this area for anything that's UTF-8 text data
     binaryData:
       label: Binary Data
+      noData: No Binary Data available for this ConfigMap...
```
Comment (richard-cox) [inline assets/translations/en-us.yaml:1717]: There should be a matching message in place for when there's no (non-binary) data. At the moment it shows the keyvalue form but with a placeholder for a row

Comment (richard-cox) [inline assets/translations/en-us.yaml:1717]: This shouldn't have the `...`

**Review by richard-cox — CHANGES_REQUESTED** [review]
I've lost track of what we were aiming for on a few things
- If there is no initial binary data the user cannot add new entries (just get the 'no binary data' message instead of `Read from Filer` button). Not sure if this was a stretch goal?
- In the config tab the user could always see binary data (in view and edit modes) instead of the summary size string (due to the configmap not setting `:display-values-as-binary="true"`). Should they see it in view but not in edit? Neil mentioned in edit mode there should be a message saying they can't edit binary data, this is missing

image
image
image

**Review by richard-cox — APPROVED** [review]

**Pull request merged**